### ISSUE TITLE
keras.ops.diagonal returns inconsistent static shape for symbolic inputs with negative offset

### ISSUE THREAD

**Issue by AGFACBNNR** [keras-team/keras#22527]
# Bug report: `keras.ops.diagonal` infers an incorrect symbolic output shape

## Environment
- Keras 3.13.2
- TensorFlow 2.21.0

## Reproduction
```python
import keras
import tensorflow as tf

x = tf.constant([[[1, 2, 3, 4],
                  [5, 6, 7, 8],
                  [9, 10, 11, 12]],
                 [[13, 14, 15, 16],
                  [17, 18, 19, 20],
                  [21, 22, 23, 24]]], dtype=tf.float32)

inp = keras.Input(shape=(3, 4))

print(keras.ops.diagonal(x, offset=-1, axis1=0, axis2=1).shape)
print(keras.ops.diagonal(inp, offset=-1, axis1=0, axis2=1).shape)
```

## Expected behavior
The eager and symbolic paths should infer the same output shape for the same input shape and arguments.

## Actual behavior
- Eager tensor output shape: `(4, 1)`
- Symbolic `keras.Input` output shape: `(4, None)`

**Comment by divyashreepathihalli** [comment]
@keras-team /triage

**Comment by amitsrivastava78** [comment]
Not a bug. The inputs have different shapes:
* Eager x → (2, 3, 4) — batch dim is 2, so diagonal length = 2 + (-1) = 1 → shape (4, 1) 
* keras.Input(shape=(3,4)) → (None, 3, 4) — batch dim is None, so diagonal length is unknown → shape (4, None) 
Fix for the user: use batch_shape to get a static shape:

inp = keras.Input(batch_shape=(2, 3, 4))
keras.ops.diagonal(inp, offset=-1, axis1=0, axis2=1).shape  # → (4, 1)

Suggest closing as not a bug / working as intended.
